src/auto-evo/simulation/food_source/HeterotrophicFoodSource.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Linq;
+
+public class HeterotrophicFoodSource : IFoodSource
+{
+    private readonly Compound oxytoxy = SimulationParameters.Instance.GetCompound("oxytoxy");
+
+    private MicrobeSpecies prey;
+    private Patch patch;
+    private float preySpeed;
+    private float preySize;
+    private float totalEnergy;
+
+    public HeterotrophicFoodSource(Patch patch, MicrobeSpecies prey)
+    {
+        this.prey = prey;
+        this.patch = patch;
+        preySpeed = prey.BaseSpeed;
+        preySize = prey.Organelles.Organelles.Sum(organelle => organelle.Definition.HexCount);
+        patch.SpeciesInPatch.TryGetValue(prey, out long population);
+        totalEnergy = population * prey.Organelles.Count * Constants.AUTO_EVO_PREDATION_ENERGY_MULTIPLIER;
+    }
+
+    public float FitnessScore(Species species)
+    {
+        var microbeSpecies = (MicrobeSpecies)species;
+
+        // No canibalism
+        if (species == prey)
+        {
+            return 0.0f;
+        }
+
+        var behaviorScore = microbeSpecies.Aggression / Constants.MAX_SPECIES_AGGRESSION;
+
+        var predatorSize = microbeSpecies.Organelles.Organelles.Sum(organelle => organelle.Definition.HexCount);
+        var predatorSpeed = microbeSpecies.BaseSpeed;
+        predatorSpeed += ProcessSystem
+            .ComputeEnergyBalance(microbeSpecies.Organelles.Organelles, patch.Biome,
+                microbeSpecies.MembraneType).FinalBalance;
+
+        // It's great if you can engulf this prey, but only if you can catch it
+        var engulfScore = predatorSize / preySize > Constants.ENGULF_SIZE_RATIO_REQ
+            && !microbeSpecies.MembraneType.CellWall ?
+                Constants.AUTO_EVO_ENGULF_PREDATION_SCORE :
+                0.0f;
+        engulfScore *= predatorSpeed > preySpeed ? 1.0f : Constants.AUTO_EVO_SLOW_ENGULF_MULTIPLIER;
```

Comment:
Although I knew the code already, I did not guess that the constant was to be used there when I saw it defined. 

It made me think of a multiplier for a slow engulf, rather than slow "engulfer"/predator. Perhaps there is a more suitable name for the constant?

Reply:
upon re-read I totally agree. Any suggestions?

Reply:
To me it really feels like as the probability of a lucky catch. `ENGULF_LUCKY_CATCH_PROBABILITY` perhaps? Or `LUCKY_ENGULF_PROBABILITY`? If you don't want to state it as a proba, you may prefer `ENGULF_CHANCE_FACTOR` or something along these lines?

Reply:
I used ENGULF_LUCKY_CATCH_PROBABILITY. Seems accurate enough.

Reply:
I assumed you'd keep the prefix by the way, although it could perhaps be generalized outside of auto evo later on.

Reply:
no no, you're right again. I'm changing it.